src/installers/GDWeaveInstaller.ts

```diff
@@ -0,0 +1,82 @@
+import { InstallArgs, PackageInstaller } from "./PackageInstaller";
+import path from "path";
+import FsProvider from "../providers/generic/file/FsProvider";
+import { addToStateFile } from "./InstallRuleInstaller";
+import FileUtils from "../utils/FileUtils";
+
+const basePackageFiles = ["manifest.json", "readme.md", "icon.png"];
+
+// Adapted from the BepInEx & Lovely installers
+export class GDWeaveInstaller implements PackageInstaller {
+    async install(args: InstallArgs) {
+        const fs = FsProvider.instance;
+        const fileRelocations = new Map<string, string>();
+
+        const winmmSrc = path.join(args.packagePath, "winmm.dll");
+        const winmmDest = args.profile.joinToProfilePath("winmm.dll");
+        await fs.copyFile(winmmSrc, winmmDest);
+        fileRelocations.set(winmmSrc, "winmm.dll");
+
+        for (const item of (await FsProvider.instance.readdir(args.packagePath))) {
+            if (basePackageFiles.includes(item.toLowerCase())) continue;
+
+            const srcPath = path.join(args.packagePath, item);
+            const destPath = args.profile.joinToProfilePath(item);
+            if ((await FsProvider.instance.stat(srcPath)).isFile()) {
+                await FsProvider.instance.copyFile(srcPath, destPath);
+            } else {
+                await FsProvider.instance.copyFolder(srcPath, destPath);
+            }
+        }
+
+        await addToStateFile(args.mod, fileRelocations, args.profile);
+    }
+}
+
+async function searchForManifest(fs: FsProvider, dir: string): Promise<string | null> {
+    if (await fs.exists(path.join(dir, "manifest.json"))) {
+        return dir;
+    }
+
+    for (const item of await fs.readdir(dir)) {
+        if ((await fs.stat(path.join(dir, item))).isDirectory()) {
+            const result = await searchForManifest(fs, path.join(dir, item));
+            if (result) return result;
+        }
+    }
+
+    return null;
+}
+
+export class GDWeavePluginInstaller implements PackageInstaller {
+    async install(args: InstallArgs) {
+        const fs = FsProvider.instance;
+        const installDir = path.join("GDWeave", "mods", args.mod.getName());
+        const files = await FsProvider.instance.readdir(args.packagePath);
+
+        // Packaging is all over the place. Find a folder with a manifest.json
```

Comment:
I'm not against this "sniffing out the mod folder" pattern as long as it's acknowledged that using it is a compromise which trades "no changes to existing packages now" to "having potential problems in the future". Providing "do it exactly like this" instructions may be less confusing for new mod authors? E.g. if someone splits their implementation to two sibling folders, only one of them will get copies. This might be unnecessary future proofing, I do that sometimes 🤷🏻

Reply:
I do have a suggested format for users, which is `GDWeave/mods/ModIDGoesHere` (documented [here](https://github.com/NotNite/GDWeave/blob/main/MODS.md#publishing-to-thunderstore)). Admittedly it would be better if it was a standard name `mod` directory like discussed prior, but I'm not sure if it's feasible to break this with over 100 packages at this point.